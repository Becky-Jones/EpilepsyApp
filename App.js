import { StatusBar } from "expo-status-bar";
import React, { useState } from "react";
import { StyleSheet, Text, View } from "react-native";
import LoginScreen from './src/main/pages/login.js'
import CreateAdmin from './src/main/pages/createAdmin.js'
import SignupScreen from './src/main/pages/createPatient.js'
import Home from './src/main/pages/home.js'
import PatientDetails from './src/main/pages/patientDetails.js'
import PersonalDetails from './src/main/pages/personalDetails.js'
import MediaDetails from './src/main/pages/mediaDetails.js'
import Analytics from './src/main/pages/analytics.js'

import AllMedia from './src/main/pages/allMedia.js'
import EditMedia from './src/main/pages/editMedia.js'
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";

const Stack = createNativeStackNavigator();

export default function App() {

  return (

  <NavigationContainer>
    <Stack.Navigator initialRouteName="Login">
      <Stack.Screen name="Login" component={LoginScreen} />
      <Stack.Screen name="Create Patient" component={SignupScreen} />
      <Stack.Screen name="Create Practitioner" component={CreateAdmin} />
      <Stack.Screen name="Home" component={Home} />
      <Stack.Screen name="Patient Details" component={PatientDetails} />
      <Stack.Screen name="Media Details" component={MediaDetails} />
      <Stack.Screen name="Analytics" component={Analytics} />
      <Stack.Screen name="All Media" component={AllMedia} />
      <Stack.Screen name="Edit Media" component={EditMedia} />
      <Stack.Screen name="Edit Personal Details" component={PersonalDetails} />
    </Stack.Navigator>
  </NavigationContainer>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    marginTop: 60,
  },

  image: {
    width: 20,
    height: 10,
    marginBottom: 40,
  },

  inputView: {
    backgroundColor: "#FFC0CB",
    borderRadius: 30,
    width: "70%",
    height: 45,
    marginBottom: 20,
    alignItems: "center",
  },

  TextInput: {
    height: 50,
    flex: 1,
    padding: 10,
    marginLeft: 20,
  },

  forgot_button: {
    height: 30,
    marginBottom: 30,
  },

  signupBtn: {
    width: "80%",
    borderRadius: 25,
    height: 50,
    alignItems: "center",
    justifyContent: "center",
    marginTop: 40,
    backgroundColor: "blue",
  },
});
