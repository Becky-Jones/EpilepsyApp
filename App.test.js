import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';

import App from './App';
import LoginScreen from './src/main/pages/login';

describe('<App />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(1);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});