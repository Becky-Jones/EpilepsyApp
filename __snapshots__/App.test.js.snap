// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={false}
        largeTitleHideShadow={false}
        title="Login"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "marginTop": 100,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 20,
                "height": 50,
                "marginBottom": 20,
                "padding": 10,
              }
            }
          >
            WCFB Epilepsy App Login
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3399ff",
                "borderRadius": 30,
                "height": 45,
                "marginBottom": 20,
                "width": "70%",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              onChangeText={[Function]}
              placeholder="Email"
              placeholderTextColor="white"
              rejectResponderTermination={true}
              style={
                Object {
                  "color": "white",
                  "flex": 1,
                  "height": 50,
                  "marginLeft": 20,
                  "padding": 10,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3399ff",
                "borderRadius": 30,
                "height": 45,
                "marginBottom": 20,
                "width": "70%",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              onChangeText={[Function]}
              placeholder="Password"
              placeholderTextColor="white"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Object {
                  "color": "white",
                  "flex": 1,
                  "height": 50,
                  "marginLeft": 20,
                  "padding": 10,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#42A5F5",
                "borderRadius": 25,
                "height": 50,
                "justifyContent": "center",
                "marginTop": 40,
                "opacity": 1,
                "width": "80%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                }
              }
            >
              LOGIN
            </Text>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
